ignite/metrics/precision.py

```diff
@@ -93,139 +126,137 @@ class Precision(_BasePrecisionRecall):
             print(state.metrics["precision"])
+            print(f"Micro Precision: {state.metrics["micro precision"]}")
+            print(f"Macro Precision: {state.metrics["macro precision"]}")
+            print(f"Weighted Precision: {state.metrics["weighted precision"]}")
+            print(f"Samples Precision: {state.metrics["samples precision"]}")
 
-        .. testoutput:: 4
+        .. testoutput:: 3
 
-            tensor([0.2000, 0.5000, 0.0000], dtype=torch.float64)
+            Precision: tensor([0.2000, 0.5000, 0.0000], dtype=torch.float64)
+            Micro Precision: 0.2222222222222222
+            Macro Precision: 0.2333333333333333
+            Weighted Precision: 0.175
+            Samples Precision: 0.2
 
-        In binary and multilabel cases, the elements of `y` and `y_pred` should have 0 or 1 values. Thresholding of
-        predictions can be done as below:
+        Thresholding of predictions can be done as below:
 
-        .. testcode:: 5
+        .. testcode:: 4
 
             def thresholded_output_transform(output):
                 y_pred, y = output
                 y_pred = torch.round(y_pred)
                 return y_pred, y
 
-            metric = Precision(average=False, output_transform=thresholded_output_transform)
+            metric = Precision(output_transform=thresholded_output_transform)
             metric.attach(default_evaluator, "precision")
             y_true = torch.Tensor([1, 0, 1, 1, 0, 1])
             y_pred = torch.Tensor([0.6, 0.2, 0.9, 0.4, 0.7, 0.65])
             state = default_evaluator.run([[y_pred, y_true]])
             print(state.metrics["precision"])
 
-        .. testoutput:: 5
+        .. testoutput:: 4
 
             0.75
 
-        In multilabel cases, average parameter should be True. However, if user would like to compute F1 metric, for
-        example, average parameter should be False. This can be done as shown below:
-
-        .. code-block:: python
-
-            precision = Precision(average=False)
-            recall = Recall(average=False)
-            F1 = precision * recall * 2 / (precision + recall + 1e-20)
-            F1 = MetricsLambda(lambda t: torch.mean(t).item(), F1)
-
-    .. warning::
-
-        In multilabel cases, if average is False, current implementation stores all input data (output and target) in
-        as tensors before computing a metric. This can potentially lead to a memory error if the input data is larger
-        than available RAM.
-
 
+    .. versionchanged:: 0.5.0
+            `average` parameter's semantic changed and three options were added to it.
     """
 
     def __init__(
         self,
         output_transform: Callable = lambda x: x,
-        average: bool = False,
+        average: Union[bool, str] = False,
         is_multilabel: bool = False,
         device: Union[str, torch.device] = torch.device("cpu"),
     ):
+
+        if average not in [False, "micro", "weighted", "samples"]:
```

Comment:
We should still support average True and translate it to appropriate str value

Reply:
Which option do you think is better to translate to? For precision, I suggest `weighted`, as it's `macro` plus considering classes/labels imbalance.

Reply:
Translation should be without any BC-breaking changes, so I'd say `macro`.

Reply:
Got it

Reply:
So, we need to add "macro" option

Reply:
Didn't we assign `average=True` to macro?